addons/l10n_it_edi_withholding/models/account_move.py

```diff
@@ -0,0 +1,97 @@
+        """Filters tax details to only include the positive amounted lines regarding VAT taxes."""
+        return tax_values['tax_repartition_line'].factor_percent >= 0 and line['taxes']._filter_kind('vat').amount >= 0
+
+    def _prepare_fatturapa_export_values(self):
+        """Add withholding and pension_fund features."""
+        template_values = super()._prepare_fatturapa_export_values()
+
+        # Withholding tax data
+        WithholdingTaxData = namedtuple('TaxData', ['tax', 'tax_amount'])
+        withholding_lines = self.line_ids.filtered(lambda x: x.tax_line_id._filter_kind('withholding'))
+        withholding_values = [WithholdingTaxData(x.tax_line_id, abs(x.balance)) for x in withholding_lines]
+
+        # Fix the total eventually as it must be without considering ritenuta
+        document_total = template_values['document_total']
+        if withholding_values:
+            document_total += sum(x.tax_amount for x in withholding_values)
+
+        # Pension fund tax data, I need the base amount so I have to sum the amounts of the lines with the tax
+        PensionFundTaxData = namedtuple('TaxData', ['tax', 'base_amount', 'tax_amount', 'vat_tax', 'withholding_tax'])
+        pension_fund_lines = self.line_ids.filtered(lambda line: line.tax_line_id._filter_kind('pension_fund'))
+        pension_fund_mapping = {}
+        for line in self.line_ids:
+            pension_fund_tax = line.tax_ids._filter_kind('pension_fund')
+            if pension_fund_tax:
+                pension_fund_mapping[pension_fund_tax.id] = (line.tax_ids._filter_kind('vat'), line.tax_ids._filter_kind('withholding'))
+
+        # Pension fund taxes in the XML must have a reference to their VAT tax (Aliquota tag)
+        pension_fund_values = []
+        enasarco_taxes = []
+        for line in pension_fund_lines:
+            # Enasarco must be treated separately
+            if line.tax_line_id.l10n_it_pension_fund_type == 'TC07':
+                enasarco_taxes.append(line.tax_line_id)
+                continue
+            pension_fund_tax = line.tax_line_id
+            # Here we are supposing that the same pension_fund is always associated to the same VAT and Withholding taxes
+            # That's also what the "Aliquota" tag seems to imply in the XML
+            vat_tax, withholding_tax = pension_fund_mapping[pension_fund_tax.id]
+            pension_fund_values.append(PensionFundTaxData(pension_fund_tax, line.tax_base_amount, abs(line.balance), vat_tax, withholding_tax))
+
+        # Enasarco pension fund must be expressed in the AltriDatiGestionali at the line detail level
+        enasarco_values = False
+        if enasarco_taxes:
+            enasarco_values = {}
+            all_tax_details = self._prepare_edi_tax_details(filter_to_apply=lambda line, tax_values: self.env['account.tax'].browse([tax_values['id']]).l10n_it_pension_fund_type == 'TC07')
+            for _key, detail in all_tax_details['tax_details_per_record'].items():
```

Comment:
why not .values() then instead of .items()?

Reply:
Yes, why?